src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java

```diff
@@ -297,40 +233,27 @@ void selfDeleteRequest_WhenPasswordWrongAndTheSameConfirmation_ShouldThrowInvali
         when(userService.isPasswordMatches(any(), any())).thenReturn(false);
 
         UserDeleteFlowDto userDeleteFlowDto = new UserDeleteFlowDto(WRONG_OLD_PASSWORD, WRONG_OLD_PASSWORD);
-        MvcResult mvcResult = getResultActions(HttpMethod.DELETE, USER_SERVICE_DELETE, userDeleteFlowDto,
-                status().isBadRequest())
-                .andDo(print())
-                .andReturn();
+        MvcResult mvcResult = sendDtoAndGetMvcResult(delete(USER_SERVICE_DELETE), userDeleteFlowDto, status().isBadRequest());
 
         assertTrue(mvcResult.getResponse().getContentAsString().contains(getMessageSource("incorrect.password")));
     }
 
-    @Test
+    @ParameterizedTest
     @WithMockUser(username = "admin")
-    void selfDeleteRequest_WhenPasswordCorrectAndConfirmationWrong_ShouldThrowIllegalArgumentException()
-            throws Exception {
-        UserDeleteFlowDto userDeleteFlowDto = new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD);
-        MvcResult mvcResult = getResultActions(HttpMethod.DELETE, USER_SERVICE_DELETE, userDeleteFlowDto,
-                status().isBadRequest())
-                .andDo(print())
-                .andReturn();
+    @MethodSource("selfDeleteTestData")
+    void selfDeleteRequest_WhenPasswordCorrectAndConfirmationWrong_ShouldThrowIllegalArgumentException(UserDeleteFlowDto testDto) throws Exception {
+        MvcResult mvcResult = sendDtoAndGetMvcResult(delete(USER_SERVICE_DELETE), testDto, status().isBadRequest());
+
         String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
 
         assertTrue(message.contains(getMessageSource("different.passwords")));
     }
 
-    @Test
-    @WithMockUser(username = "admin")
-    void selfDeleteRequest_WhenPasswordWrongAndPasswordConfirmationDoesNotMatch_ShouldThrowIllegalArgumentException()
-            throws Exception {
-        UserDeleteFlowDto userDeleteFlowDto = new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD);
-        MvcResult mvcResult = getResultActions(HttpMethod.DELETE, USER_SERVICE_DELETE, userDeleteFlowDto,
-                status().isBadRequest())
-                .andDo(print())
-                .andReturn();
-        String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
-
-        assertTrue(message.contains(getMessageSource("different.passwords")));
+    private static Stream<Arguments> selfDeleteTestData() {
+        return Stream.of(
+                Arguments.of(new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD)),
+                Arguments.of(new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD))
```

Comment:
Why do we need the same data here (new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD)) ?

Reply:
😄 two tests which existed before had the same values and actually did the same, I only refactored and was pretty tired to find the difference into the data. Will be changed, thanks.